src/server/journal/serializer.cc

```diff
@@ -134,36 +138,36 @@ template io::Result<uint16_t> JournalReader::ReadUInt<uint16_t>();
 template io::Result<uint32_t> JournalReader::ReadUInt<uint32_t>();
 template io::Result<uint64_t> JournalReader::ReadUInt<uint64_t>();
 
-io::Result<size_t> JournalReader::ReadString() {
+io::Result<size_t> JournalReader::ReadString(std::string* command_buf) {
   size_t size = 0;
   SET_OR_UNEXPECT(ReadUInt<uint64_t>(), size);
 
   if (auto ec = EnsureRead(size); ec)
     return make_unexpected(ec);
 
-  unsigned offset = str_buf_.size();
-  str_buf_.resize(offset + size);
-  buf_.ReadAndConsume(size, str_buf_.data() + offset);
+  unsigned offset = command_buf->size();
+  command_buf->resize(offset + size);
+  buf_.ReadAndConsume(size, command_buf->data() + offset);
 
   return size;
 }
 
-std::error_code JournalReader::Read(CmdArgVec* vec) {
+std::error_code JournalReader::ReadCommand(journal::ParsedEntry::CmdData* data) {
   size_t num_strings = 0;
   SET_OR_RETURN(ReadUInt<uint64_t>(), num_strings);
-  vec->resize(num_strings);
+  data->cmd_args.resize(num_strings);
 
   // Read all strings consecutively.
-  str_buf_.clear();
-  for (auto& span : *vec) {
```

Comment:
Previously, we had a str_buf_ that was part of the reader and had some pre-allocated size, now we resize the command buf for every string we read with possible re-allocations.

Changing it should be very simple, just keep str_buf_, and when setting span pointers, copy it into command_buf

Reply:
true, but also if I use str_buf_ we will do copy from socket to buf_, from buf_ to  str_buf_ and than resize data->command_buf and copy  str_buf_  to data->command_buf.
I will change that the journal writer to write the command string length and by this we can just do resize data->command_buf, we will not need str_buf_ and also we will not need 2 loops , one for filling the size in span and one for pointer, we will just have one loop.

Reply:
Thats also true, but requires changing the format 😄
Btw, instead of writing the number of slices, we can just write the total length and then read until we reached it. But we can add this small optimization later if its too much for now